linetools/guis/line_widgets.py

```diff
@@ -0,0 +1,298 @@
+""" Module for line widets
+"""
+from __future__ import print_function, absolute_import, division, unicode_literals
+
+from PyQt4 import QtGui
+from PyQt4 import QtCore
+
+import numpy as np
+
+from astropy.table import Table
+
+from linetools.guis import utils as ltgu
+from linetools.lists.linelist import LineList
+
+# #####
+class PlotLinesWidget(QtGui.QWidget):
+    """ Widget to set up spectral lines for plotting
+    """
+    def __init__(self, parent=None, status=None, init_llist=None, init_z=None):
+        """
+        Parameters
+        ----------
+        parent
+        status
+        init_llist
+        init_z
+
+        Returns
+        -------
+
+        """
+        super(PlotLinesWidget, self).__init__(parent)
+
+        # Initialize
+        if not status is None:
+            self.statusBar = status
+        if init_z is None:
+            init_z = 0.
+
+        # Create a dialog window for redshift
+        z_label = QtGui.QLabel('z=')
+        self.zbox = QtGui.QLineEdit()
+        self.zbox.z_frmt = '{:.7f}'
+        self.zbox.setText(self.zbox.z_frmt.format(init_z))
+        self.zbox.setMinimumWidth(50)
+        self.connect(self.zbox, QtCore.SIGNAL('editingFinished ()'), self.setz)
+
+        # Create the line list
+        self.lists = ['None', 'ISM', 'Strong', 'Galaxy', 'H2', 'EUV', 'OVI']
+        #'grb.lst', 'dla.lst', 'lls.lst', 'subLLS.lst',
+#                      'lyman.lst', 'Dlyman.lst', 'gal_vac.lst', 'ne8.lst',
+#                      'lowz_ovi.lst', 'casbah.lst', 'H2.lst']
+        list_label = QtGui.QLabel('Line Lists:')
+        self.llist_widget = QtGui.QListWidget(self)
+        for ilist in self.lists:
+            self.llist_widget.addItem(ilist)
+        self.llist_widget.setCurrentRow(0)
+        self.llist_widget.currentItemChanged.connect(self.on_list_change)
+        self.llist_widget.setMaximumHeight(100)
+
+        # Input line list?
+        if init_llist is None:
+            self.llist = {} # Dict for the line lists
+            self.llist['Plot'] = False
+            self.llist['z'] = 0.
+            self.llist['List'] = 'None'
+            self.llist['Lists'] = []
+        else: # Fill it all up and select
+            self.llist = init_llist
+            if not init_llist['List'] in self.lists:
+                self.lists.append(init_llist['List'])
+                self.llist_widget.addItem(init_llist['List'])
+                self.llist_widget.setCurrentRow(len(self.lists)-1)
+            else:
+                idx = self.lists.index(init_llist['List'])
+                self.llist_widget.setCurrentRow(idx)
+            try:
```

Comment:
Please have the 'select' part of the code (i.e. current lines 75 and 76) in an independent block, after the `self.linelist` is fully initialized. You can then remove current line 73.